### DISCUSSION TITLE
Multi-provider retrievals

### DISCUSSION THREAD

**Discussion by RobQuistNL** [filecoin-project/boost#545] [Client]
### Checklist

- [X] This is **not** a new feature or an enhancement to the Filecoin protocol. If it is,  please open an [FIP issue](https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0001.md).
- [X] This is **not** brainstorming ideas. If you have an idea you'd like to discuss, please open a new discussion on [the lotus forum](https://github.com/filecoin-project/lotus/discussions/categories/ideas) and select the category as `Ideas`.
- [X] I **have** a specific, actionable, and well motivated feature request to propose.

### Lotus component

- [ ] lotus daemon - chain sync
- [ ] lotus miner - mining and block production
- [ ] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt)
- [ ] lotus miner/market - storage deal
- [X] lotus miner/market - retrieval deal
- [X] lotus miner/market - data transfer
- [ ] lotus client
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [ ] Other

### What is the motivation behind this feature request? Is your feature request related to a problem? Please describe.

Since we're working on Evergreen, and not every host is really quick, it would be really befinicial if we could retrieve data from multiple miners at the same time.

### Describe the solution you'd like

`lotus client retrieve --providers f0773157,f01240,f08403,f010617,f01611097 --car 'uAXCg5AIgvSmwsCk0K1OiOtB1uhfd8kS25R6wrLnEEVMXHt8Lz3w'` to retrieve from multiple sources at once

### Describe alternatives you've considered

There's not really a way for me to do a partial retrieval (if I'm not mistaken) and then rebuild the result - that would also work.

### Additional context

Multiple retrievals == more speed == more better :)

**Comment by dirkmc** [comment]
This is currently not possible with graphsync retrieval. We would need to build a bitswap interface for retrieval on the server side.
